bgpd/bgp_network.c

```diff
@@ -284,6 +286,10 @@ static int bgp_accept(struct thread *thread)
 		return -1;
 	}
 	listener->thread = NULL;
+
+	if (listener->bgp)
```

Comment:
This assignment makes no sense.  Immediately below we call bgp_get_instance_for_inc_conn with the address of &bgp, which in turn immediately NULLS it out.  Why do we need this?